### DISCUSSION TITLE
Updating FBM stock via SP-API

### DISCUSSION THREAD

**Discussion by JesusR25** [amzn/selling-partner-api-models#4388] [Q&A]
Hello again everyone! Is there a way to update the FBM stock? On most of my listings of that type, I no longer have stock, and I would like to update the stock via the SP-API. Is this possible? Thanks to everyone!

**Comment by mafge** [comment]
Hi @JesusR25,
thanks for reaching out!
Yes, you can update stock via the API. You have to use `patchListingsItem` from [Listings Items API](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#patchlistingsitem). You can find detailed instructions in the [use case guide](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-use-case-guide#tutorial-partially-update-a-listing). You can use the following JSON snippet in your request:
```
{
    "productType":
    "PRODUCT",
    "patches": [{
        "op": "replace",
        "path": "/attributes/fulfillment_availability",
        "value": [{
            "fulfillment_channel_code": "DEFAULT",
            "quantity": 0 <-- Enter your current stock level here
        }]
    }]
}
```
Hope that helps!
Best,
Marc
Selling Partner Developer Services
